srcStatic/ModuleLoader.cpp

```diff
@@ -105,8 +158,19 @@ void ModuleLoader::RegisterModule(std::unique_ptr<GameModule> newGameModule)
 void ModuleLoader::LoadModules()
 {
 	RegisterBuiltInModules();
+	RegisterConsoleModules();
 	RegisterIniModules();
 
+	// Abort early to avoid hooking file search path if not needed
+	if (modules.empty()) {
+		return;
+	}
+
+	// Setup loading of additional resources from module folders
+	ResourceSearchPath::Activate();
```

Comment:
Looks like the `ResourceSearchPath` is now being activated if any modules are loaded, not just if any console modules are loaded. This may pose a problem on Mingw, as the current hook tries to chain to the original Outpost2.exe method when the augmented path doesn't find a file. That chaining to the original method was affected by a Mingw code gen bug. This will need some additional testing.

Reply:
Now that unit tests are passing, I gave this a try. The game does indeed crash now on Linux if any modules are loaded. Previously it only crashed if `/loadmod` was used. Now it crashes all the time since there are BuiltIn modules or IniModules that get loaded and cause `ResourceSearchPath::Activate()` to be called.

Maybe that call can be moved into the Console module loading code for now, and made conditional on console modules being present.

----

Side note: Initially, I was confused by this not crashing as I expected. This was until I realized I was testing on my old stock INI file, but recent changes have renamed certain keys used for module loading. I had to create an "ExternalModules" key to get INI file loading working again before I could see this crash. This is a reminder that although INI keys might be renamed in this project, the game is packaged separately, and corresponding updates would need to be made there.

Reply:
Update: If you merge PR #196 first, it should mitigate this issue.